api/api/utils/generate_teams_data_loader.py

```diff
@@ -1,40 +1,76 @@
-from typing import Dict, Any, List
-from dataclasses import dataclass, field
+from typing import Dict, Any, List, Optional
+from dataclasses import dataclass
+
+from schema import Schema
 
 from api.ai.new.interfaces.algorithm_options import AlgorithmOptions
 from api.ai.new.interfaces.team_generation_options import TeamGenerationOptions
 from api.models.enums import AlgorithmType
-from api.models.project import Project
 from api.models.student import Student
+from api.api.utils.algorithm_option import get_algorithm_option_class
 
 
 @dataclass
 class GenerateTeamsInputData:
     students: List[Student]
     algorithm_type: AlgorithmType
-    algorithmOptions: AlgorithmOptions
+    algorithm_options: AlgorithmOptions
     team_generation_options: TeamGenerationOptions
-    projects: List[Project] = field(default_factory=list)
 
 
 class GenerateTeamsDataLoader:
     def __init__(self, validated_data: Dict[str, Any]):
         self.data = validated_data
 
     def load(self) -> GenerateTeamsInputData:
-        pass
+        return GenerateTeamsInputData(
+            students=self._get_students(),
+            team_generation_options=self._get_team_generation_options(),
+            algorithm_type=self._get_algorithm_type(),
+            algorithm_options=self._get_algorithm_options(),
+        )
 
     def _get_algorithm_type(self) -> AlgorithmType:
-        pass
+        algorithm_options = self.data.get("algorithm_options")
+        algorithm_type = algorithm_options.get("algorithm_type")
+        return AlgorithmType(algorithm_type)
 
     def _get_students(self) -> List[Student]:
-        pass
-
-    def _get_projects(self) -> List[Project]:
-        pass
+        students = self.data.get("students")
+        return [
+            Student(
+                _id=student.get("id"),
+                name=student.get("name"),
+                relationships=student.get("relationships"),
+                attributes=student.get("attributes"),
+                project_preferences=student.get("project_preferences"),
+            )
+            for student in students
+        ]
 
     def _get_algorithm_options(self) -> AlgorithmOptions:
-        pass
+        algorithm_options = self.data.get("algorithm_options")
+        algorithm_type = AlgorithmType(algorithm_options.pop("algorithm_type"))
+        algorithm_option_class = get_algorithm_option_class(algorithm_type)
+        return algorithm_option_class.parse_json(algorithm_options)
+
+    def _validate_team_generation_options(
+        self, team_generation_options: Optional[Dict[str, Any]]
+    ):
+        if team_generation_options is None:
+            raise ValueError("team_generation_options is required")
+        Schema(int).validate(team_generation_options.get("total_teams"))
+        Schema(list).validate(team_generation_options.get("initial_teams"))
```

Comment:
leave a todo or ticket because this validation is not as simple as just being a list